### PR TITLE
Add configuration to default dashboard

### PR DESCRIPTION
**Pull request by piitaya**
### Strategy editor

Adds UI editor for strategies. The user can switch between UI and yaml to edit the strategy configuration. If the strategy doesn't provide editor, it will be yaml only.

### Default dashboard improvements

Add 3 options to default dashboard : 
- **hidden_areas** : List of areas that must not be displayed
-  **hide_entities_without_area**: Entities without area will not be listed
-  **hide_energy**: Hide the energy card

### Demo

https://github.com/home-assistant/frontend/assets/5878303/9ded0d98-91a6-459e-81d8-37325db1d14b

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by balloob** [comment]
Instead of hidden areas, could we have a single area selector component that allows:

- enable/disable area (current hide functionality)
- re-order area

I am sure that this will become useful for many other strategies too.

**Comment by piitaya** [comment]
Yep definitely! We already discussed about a better UX for area during frontend meeting 🙂

**Comment by balloob** [comment]
Yeah that sounds about right, and I guess that means that we can also add that in a future PR. Should we then already adopt the YAML format that you just proposed to avoid having to overload `hidden_areas` key? (assuming selector only works on 1 key)

File: src/translations/en.json [hunk 1]
```diff
@@ -4863,6 +4863,13 @@
               }
             }
           },
+          "strategy-editor": {
+            "header": "Edit dashboard",
+            "take_control": "Take control",
+            "raw_configuration_editor": "[%key:ui::panel::lovelace::editor::menu::raw_editor%]",
+            "show_code_editor": "[%key:ui::panel::lovelace::editor::edit_card::show_visual_editor%]",
```
Comment (silamon) [inline src/translations/en.json:4870]:
```suggestion
            "show_code_editor": "[%key:ui::panel::lovelace::editor::edit_card::show_code_editor%]",
```

Comment (piitaya) [reply src/translations/en.json:4870]: Good catch! Thanks

**Review by bramkragten — APPROVED** [review]

**Pull request merged**